napari_aicssegmentation/controller/workflow_select_controller.py

```diff
@@ -75,7 +75,12 @@ def select_workflow(self, workflow_name: str):
             + workflow_name,
         )
 
-        self.model.active_workflow = self._workflow_engine.get_executable_workflow(workflow_name, layer0.data)
+        # TODO: Make sure the channel being selected is the correct dimension (when we start using AICSIMAGEIO)
+
+        self.model.active_workflow = self._workflow_engine.get_executable_workflow(
+            workflow_name, LayerReader.get_selected_channel(LayerReader,
```

Comment:
I'm amazed Python allows to do this - but this is not how we should be using objects. You're using an instance method like a class method here by manually passing the `self` argument. This is an anti pattern. We either should be defining a method as `@classmethod` or `@staticmethod` and use it as such - or use instance methods as they are meant to be used - from an instance.
In this case, just pass in a LayerReader instance to this controller if it's not already available. See this https://github.com/AllenCell/napari-aicssegmentation/blob/main/napari_aicssegmentation/controller/workflow_select_controller.py#L16 and this https://github.com/AllenCell/napari-aicssegmentation/blob/main/napari_aicssegmentation/core/router.py#L23.

Reply:
Keep in mind that instance methods are easier to work with when unit testing